openam-ui/openam-ui-ria/src/test/js/org/forgerock/openam/ui/common/util/PromiseTest.js

```diff
@@ -48,97 +57,120 @@ define([
                         expect(resolvedSpy).to.be.calledWith([1]);
                     });
-                    it("groups multiple resolved values into an array", () => {
+                    it("groups multiple resolved values into an array", async () => {
                         d.resolve(1, 2, 3);
+
+                        await all;
                         expect(resolvedSpy).to.be.calledWith([[1, 2, 3]]);
                     });
-                    it("rejects the returned promise if the passed in promise is rejected", () => {
-                        d.reject();
-                        expect(resolvedSpy).to.not.be.called;
-                        expect(rejectedSpy).to.be.called;
-                    });
                 });
                 context("of 2 promises", () => {
                     let d1;
                     let d2;
-                    let p;
+                    let all;
                     let resolvedSpy;
                     let rejectedSpy;
 
                     beforeEach(() => {
                         d1 = $.Deferred();
                         d2 = $.Deferred();
-                        p = Promise.all([d1.promise(), d2.promise()]);
                         resolvedSpy = sinon.spy();
                         rejectedSpy = sinon.spy();
-                        p.then(resolvedSpy, rejectedSpy);
+                        all = Promise.all([d1.promise(), d2.promise()]).then(resolvedSpy, rejectedSpy);
                     });
                     it("doesn't resolve the returned promise if neither of the promises are resolved", () => {
                         expect(resolvedSpy).to.not.be.called;
                         expect(rejectedSpy).to.not.be.called;
                     });
-                    it("doesn't resolve the returned promise if only the first promise is resolved", () => {
+                    // eslint-disable-next-line max-len
+                    it("doesn't resolve the returned promise if only the first promise is resolved", async function () {
                         d1.resolve();
+
+                        await d1.promise();
+                        // Make sure spies are not checked too early
+                        this.timeout(2000);
+                        await waitFor($, 1500);
                         expect(resolvedSpy).to.not.be.called;
                         expect(rejectedSpy).to.not.be.called;
                     });
-                    it("doesn't resolve the returned promise if only the second promise is resolved", () => {
+                    // eslint-disable-next-line max-len
+                    it("doesn't resolve the returned promise if only the second promise is resolved", async function () {
                         d2.resolve();
+
+                        await d2.promise();
+                        // Make sure spies are not checked too early
+                        this.timeout(2000);
+                        await waitFor($, 1500);
```

Comment:
This is very strange construct. Why do we need to wait? Also the "await all" construct seems strange as well. It should be possible to simply await the promise we want to be resolved, or am I missing something?

Reply:
Rewritten the "waiting" part with setTimeout.

The "all" variable is the promise returned by `Promise.all`, that's the promise we want to test here I believe. Therefore we need wait till it's resolved and then we can call the assert functions.